plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -0,0 +1,187 @@
+            description: 'The description of the merge request',
+          },
+          targetPath: {
+            type: 'string',
+            title: 'Repository Subdirectory',
+            description: 'Subdirectory of repository to apply changes to',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          projectid: {
+            title: 'Gitlab Project id',
+            type: 'string',
+          },
+          mergeRequestURL: {
+            title: 'MergeRequest(MR) URL',
+            type: 'string',
+            description: 'Link to the merge request in GitLab',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const repoUrl = ctx.input.repoUrl;
+      const { host } = parseRepoUrl(repoUrl, integrations);
+      const integrationConfig = integrations.gitlab.byHost(host);
+
+      const actions: CommitAction[] = [];
+
+      const destinationBranch = ctx.input.branchName;
+
+      if (!integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+
+      if (!integrationConfig.config.token) {
+        throw new InputError(`No token available for host ${host}`);
+      }
+
+      const api = new Gitlab({
+        host: integrationConfig.config.baseUrl,
+        token: integrationConfig.config.token,
+      });
+
+      const fileRoot = ctx.workspacePath;
+      const localFilePaths = await globby([`${ctx.input.targetPath}/**`], {
+        cwd: fileRoot,
+        gitignore: true,
+        dot: true,
+      });
+
+      const fileContents = await Promise.all(
+        localFilePaths.map(p => readFile(resolveSafeChildPath(fileRoot, p))),
+      );
+
+      const repoFilePaths = localFilePaths.map(repoFilePath => {
+        return repoFilePath;
+      });
+
+      for (let i = 0; i < repoFilePaths.length; i++) {
+        actions.push({
+          action: 'create',
+          filePath: repoFilePaths[i],
+          content: fileContents[i].toString(),
+        });
+      }
+
+      const projects = await api.Projects.show(ctx.input.projectid);
+
+      const { default_branch: defaultBranch } = projects;
+
+      try {
+        await api.Branches.create(
+          ctx.input.projectid,
+          destinationBranch,
+          String(defaultBranch),
+        ).then(branchResponse => {
+          return branchResponse;
+        });
+      } catch (e) {
+        throw new InputError(`The branch creation failed ${e}`);
+      }
+
+      try {
+        await api.Commits.create(
+          ctx.input.projectid,
+          destinationBranch,
+          ctx.input.title,
+          actions,
+        );
+      } catch (e) {
+        throw new InputError(
+          `Committing the changes to ${destinationBranch} failed ${e}`,
+        );
+      }
+
+      try {
+        const mergeRequestUrl: any = await api.MergeRequests.create(
```

Comment:
Is it possible to get rid of this `any` as well? They're a bit of a trap that might lead to future or existing bugs